extensions/amp-experiment/0.1/variant.js

```diff
@@ -32,20 +35,38 @@ export function allocateVariant(win, config) {
   const cidScope =
       config.cidScope === undefined ? 'amp-experiment' : config.cidScope;
 
-  return getBucketTicket(win, cidScope).then(bucketTicket => {
-    let upperBound = 0;
+  let hasConsentPromise = Promise.resolve(true);
 
-    // Loop through keys in a specific order since the default object key
-    // enumeration is implementation (browser) dependent.
-    const variantNames = Object.keys(config.variants).sort();
-    for (let i = 0; i < variantNames.length; i++) {
-      upperBound += config.variants[variantNames[i]];
-      if (bucketTicket < upperBound) {
-        return variantNames[i];
-      }
-    }
-    return null;
-  });
+  if (cidScope && config.consentNotificationId) {
+    hasConsentPromise = userNotificationManagerFor(win)
+        .then(manager => manager.get(config.consentNotificationId))
+        .then(userNotification => {
+          user.assert(userNotification,
+              `Notification not found: ${config.consentNotificationId}`);
+          return userNotification.isDismissed();
+        });
+  }
+
+  return hasConsentPromise
+      .then(hasConsent => {
+        if (!hasConsent) {
+          return null;
```

Comment:
I'm not sure we want to return `null` here. Imagine the user hasn't dismissed the consent notification yet, but will eventually. We're specifically not allocating this user into a variant since we can't get consent (we can't render block forever). But this is distinct from allocating the user into the `null` variant below (because the publisher didn't provide a full 100% distribution of variants).